bolt12-tip-jar/src/main/scala/fr/acinq/eclair/plugins/tipjar/TipJarPlugin.scala

```diff
@@ -38,23 +45,53 @@ class TipJarPlugin extends Plugin with RouteProvider with Logging {
   }
 
   override def onSetup(setup: Setup): Unit = {
+    val intermediateNodes = if (setup.config.hasPath("tip-jar.intermediate-nodes")) {
+      setup.config.getStringList("tip-jar.intermediate-nodes").asScala.toSeq.map(s => PublicKey(ByteVector.fromValidHex(s)))
+    }
+    else {
+      Nil
+    }
+    val dummyHops = if (setup.config.hasPath("tip-jar.dummy-hops")) {
+      setup.config.getInt("tip-jar.dummy-hops")
+    }
+    else {
+      0
+    }
     config = TipJarConfig(
       setup.config.getString("tip-jar.description"),
       MilliSatoshi(setup.config.getLong("tip-jar.default-amount-msat")),
-      CltvExpiryDelta(setup.config.getInt("tip-jar.max-final-expiry-delta")))
+      CltvExpiryDelta(setup.config.getInt("tip-jar.max-final-expiry-delta")),
+      intermediateNodes,
+      dummyHops)
   }
 
   override def onKit(kit: Kit): Unit = {
-    val tipJarHandler = kit.system.spawn(Behaviors.supervise(TipJarHandler(kit.nodeParams.nodeId, config.defaultAmount, config.maxFinalExpiryDelta)).onFailure(SupervisorStrategy.restart), "tip-jar-handler")
-    val offer = Offer(None, config.offerDescription, kit.nodeParams.nodeId, Features.empty, kit.nodeParams.chainHash)
-    kit.offerManager ! RegisterOffer(offer, kit.nodeParams.privateKey, None, tipJarHandler)
+    require(kit.nodeParams.features.hasFeature(Features.RouteBlinding))
+    require(kit.nodeParams.features.hasFeature(Features.OnionMessages))
+    val channelFees = kit.nodeParams.relayParams.publicChannelFees
+    val dummyHops = Seq.fill(config.dummyHops)(DummyBlindedHop(channelFees.feeBase, channelFees.feeProportionalMillionths, kit.nodeParams.channelConf.expiryDelta))
+    val route = ReceivingRoute(config.intermediateNodes :+ kit.nodeParams.nodeId, config.maxFinalExpiryDelta, dummyHops)
+    val tipJarHandler = kit.system.spawn(Behaviors.supervise(TipJarHandler(route, config.defaultAmount)).onFailure(SupervisorStrategy.restart), "tip-jar-handler")
+    val (blindingRouteTlvs, pathId_opt, key) = if (config.intermediateNodes.nonEmpty || config.dummyHops > 0) {
+      val pathId = randomBytes32()
+      val path = OnionMessages.buildRoute(
+        randomKey(),
+        config.intermediateNodes.map(IntermediateNode(_)) ++ Seq.fill(config.dummyHops)(IntermediateNode(kit.nodeParams.nodeId)),
+        Recipient(kit.nodeParams.nodeId, Some(pathId)))
+      (Set[OfferTlv](OfferPaths(Seq(path))), Some(pathId), randomKey())
+    } else {
+      (Set.empty[OfferTlv], None, kit.nodeParams.privateKey)
+    }
```

Comment:
nit: it would be easier to read if this `if/else` returned the offer itself. This way the case where we use a blinded path would use the `withPaths` offer constructor while the `else` would use the default constructor.

Reply:
Done, I've aligned it with the latest changes in eclair. The description is now optional and it generates a deterministic offer like lightning-kmp does.